onnxruntime/core/platform/windows/logging/etw_sink.cc

```diff
@@ -147,11 +147,11 @@ void EtwRegistrationManager::LazyInitialize() {
   if (!initialized_) {
     std::lock_guard<OrtMutex> lock(init_mutex_);
     if (!initialized_) {  // Double-check locking pattern
```

Comment:
Why don't we just use a function local static variable for doing all these things? Without the need of manually handling Double-check locking pattern

Reply:
Not sure I got your point, the EtwRegistrationManger is a static variable and requires initialization, how can we make the initialization static instead of locking?

Reply:
```
static etw_status_ = ::TraceLoggingRegisterEx(etw_provider_handle, ORT_TL_EtwEnableCallback, nullptr);
```
Then remove all the locks and the initialized_ variable. See how CPUExecutionProvider::GetKernelRegistry() works.